packages/tempus/src/Tempus_StepperExplicit_decl.hpp

```diff
@@ -0,0 +1,137 @@
+// @HEADER
+// ****************************************************************************
+//                Tempus: Copyright (2017) Sandia Corporation
+//
+// Distributed under BSD 3-clause license (See accompanying file Copyright.txt)
+// ****************************************************************************
+// @HEADER
+
+#ifndef Tempus_StepperExplicit_decl_hpp
+#define Tempus_StepperExplicit_decl_hpp
+
+// Tempus
+#include "Tempus_Stepper.hpp"
+
+
+namespace Tempus {
+
+
+/** \brief Thyra Base interface for implicit time steppers.
+ *
+ */
+template<class Scalar>
+class StepperExplicit : virtual public Tempus::Stepper<Scalar>
+{
+public:
+
+  /// \name Basic explicit stepper methods
+  //@{
+    virtual void setModel(
+      const Teuchos::RCP<const Thyra::ModelEvaluator<Scalar> >& appModel);
+    virtual void setNonConstModel(
+      const Teuchos::RCP<Thyra::ModelEvaluator<Scalar> >& appModel);
+    virtual Teuchos::RCP<const Thyra::ModelEvaluator<Scalar> >
+      getModel(){return appModel_;}
+
+    virtual Scalar getInitTimeStep(
+        const Teuchos::RCP<SolutionHistory<Scalar> >& solutionHistory) const
+      {return std::numeric_limits<Scalar>::max();}
+
+    /// Set the initial conditions, make them consistent, and set needed memory.
+    virtual void setInitialConditions (
+      const Teuchos::RCP<SolutionHistory<Scalar> >& solutionHistory){}
+
+    /// Set solver via ParameterList solver name.
+    virtual void setSolver(std::string solverName);
+    /// Set solver via solver ParameterList.
+    virtual void setSolver(
+      Teuchos::RCP<Teuchos::ParameterList> solverPL=Teuchos::null);
+    /// Set solver.
+    virtual void setSolver(
+      Teuchos::RCP<Thyra::NonlinearSolverBase<Scalar> > solver);
+    virtual Teuchos::RCP<Thyra::NonlinearSolverBase<Scalar> > getSolver() const
+      { return Teuchos::null; }
+
+    virtual std::string getStepperType() const
+     { return stepperPL_->get<std::string>("Stepper Type"); }
+
+    /// Pass initial guess to Newton solver (only relevant for implicit solvers)
+    //  thus a no-op for explicit steppers.
+    virtual void setInitialGuess(
+      Teuchos::RCP<const Thyra::VectorBase<Scalar> > initial_guess){}
+
+    virtual bool isExplicit()         const {return true;}
+    virtual bool isImplicit()         const {return false;}
+    virtual bool isExplicitImplicit() const
+      {return isExplicit() and isImplicit();}
+    virtual bool isOneStepMethod()    const {return true;}
+    virtual bool isMultiStepMethod()  const {return !isOneStepMethod();}
+
+    virtual bool getEmbedded() const
+      { return stepperPL_->get<bool>("Use Embedded", false); }
+    virtual void setUseFSAL(bool a) {stepperPL_->set<bool>("Use FSAL", a);}
+    virtual bool getUseFSAL() const {return stepperPL_->get<bool>("Use FSAL");}
```

Comment:
should return a default value of `false`

Reply:
Got it!